presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -439,4 +449,41 @@ public Type getIntermediateType()
         }
         return types.build();
     }
+
+    private class ProcessPageWork
+            implements Work<Void>
+    {
+        private final Page page;
+        private final Work<GroupByIdBlock> prerequisite;
```

Comment:
I see two alternatives here:
* Model this as a general purpose TranformWork (like `Futures.transform` in guava) that takes a `Work<I>`, and a `Function<I,O>`. In this case, calling it `prerequisite` totally makes sense.
* Model this specifically for this use case. In such a case, `prerequisite` isn't a great name. You will also want to find a better name for `ProcessPageWork` is too vague and meaningless.

If the first works out, I think it's a better option. I'll leave the class as a static inner class for now. It can be moved to a better place for utility classes if it turns out to be necessary.